### PR TITLE
macos build fix

### PR DESCRIPTION
**Pull request by gurinderu**
Small fixes for new macos sierra.

### CODE REVIEW

**Comment by mgaudet** [comment]
Thanks for these fixes. There's a couple of things we'll need before we can accept them. First and foremost, commits have to be signed by an email that has signed the Eclipse Contributors Agreement. See CONTRIBUTING.md for more details. 

Second, I'd prefer to have the changes in ` thread/unix/rasthrsup.c ` and `gc/base/CollectorLanguageInterface.cpp ` split into their own commits, with better messages about what the fixes are. In particular in `gc/base/CollectorLanguageInterface.cpp` it's not clear to me why adding the new guards is the right choice -- I worry all it does is hide those bodies from OS/X tests.

File: gc/base/CollectorLanguageInterface.cpp [hunk 1]
```diff
@@ -31,28 +31,33 @@ struct OMR_VMThread;
 void
 MM_CollectorLanguageInterface::writeBarrierStore(OMR_VMThread *omrThread, omrobjectptr_t parentObject, fomrobject_t *parentSlot, omrobjectptr_t childObject)
 {
-	GC_SlotObject slotObject(omrThread->_vm, parentSlot);
-	slotObject.writeReferenceToSlot(childObject);
```
Comment (charliegracie) [inline gc/base/CollectorLanguageInterface.cpp]: I just ran into this problem yesterday.  I think a better solution would be to just pass the OMR_VMThread through to generationalWriteBarrierStore and concurrentWriteBarrierStore as the first parameter and fetch the MM_EnvironmentStandard inside of those calls.  I think it cleans up the API and makes it easier for them to be called from other locations.

**Comment by gurinderu** [comment]
@mgaudet Maybe better to do 2 PRs for this? 
First for fix unused variables and second for fix call osx deprecated API?

**Comment by gurinderu** [comment]
I am going to make two separates PRs

**Pull request closed without merging**